src/DataStructures/DataBox/Prefixes.hpp

```diff
@@ -14,32 +16,122 @@
 #include "Utilities/TMPL.hpp"
 #include "Utilities/TypeTraits.hpp"
 
+/// \cond
 namespace Tags {
-/// \ingroup DataBoxTagsGroup
-/// \brief prefix for a `VolumeDim` spatial derivatives with respect to the `Fr`
-/// coordinate frame.
 template <typename Tag, typename VolumeDim, typename Fr,
           typename = std::nullptr_t>
 struct d;
+template <size_t Dim>
+class Extents;
+}  // namespace Tags
 
-template <typename Tag, typename VolumeDim, typename Fr>
-struct d<Tag, VolumeDim, Fr, Requires<tt::is_a_v<Tensor, db::item_type<Tag>>>>
-    : db::DataBoxPrefix {
+class DataVector;
+template <size_t Dim>
+class Index;
+template <class>
+class Variables;
+
+template <typename DerivativeTags, typename VariableTags, size_t Dim>
```

Comment:
Having to include these forward declarations suggests to me that we might want to move the compute item tags to where the function is.  Thoughts?

Reply:
I think there are sort of two options for other locations:

1. A tag file closer to the function, but we still forward declare the functions. This will reduce annoying compilation dependencies that will otherwise crop up. That is, if you only need the Tag but don't need the function definition you shouldn't need the function's header (for function templates only is this a concern)

2. We add the Tag to the header of the function. For functions where we have explicit instantiations and the header file does not contain a definition this is probably ideal. For function templates where we can't generate explicit instantiations this could be less optimal.

Have said all that, only functions that act on Variables will we not be able to generate explicit instantiations trivially. However, there aren't too many of these, and what we can do is have the definition in a cpp file, that cpp file then gets included in a cpp file for the system, which generates the explicit instantiations and all problems are solved. 2 + generating instantiations for specific systems by putting the responsibility on that system's author is best.

Reply:
I think I like the last suggestion best; but I'm happy to hear other opinions

Reply:
I agree, I like 2.